fastscanner/adapters/realtime/polygon_realtime.py

```diff
@@ -0,0 +1,153 @@
+            feed=Feed.RealTime,
+            market=Market.Stocks,
+        )
+        self._running = True
+        logger.info("Connecting WebSocket")
+
+        self._ws_task = asyncio.create_task(self._client.connect(self.handle_messages))
+
+    async def stop(self):
+        if not self._running:
+            logger.warning("WebSocket is not running.")
+            return
+        if self._client is None:
+            raise RuntimeError("WebSocketClient is None during stop.")
+
+        if self._symbols:
+            await self.unsubscribe(self._symbols)
+
+        try:
+            await self._client.close()
+            if self._ws_task:
+                await self._ws_task
+        except ConnectionClosedError as e:
+            logger.warning(f"Client was disconnected: {e}")
+        finally:
+            self._running = False
+            logger.info("WebSocket stopped.")
+
+    async def subscribe(self, symbols: set[str]):
+        if not self._running:
+            logger.warning("WebSocket is not running")
+            return
+        if self._client is None:
+            raise RuntimeError("WebSocketClient is None during subscribe.")
+
+        if not symbols:
+            logger.warning("No symbols to subscribe.")
+            return
+
+        tickers = [f"AM.{symbol}" for symbol in symbols]
+        self._client.subscribe(*tickers)
+        self._symbols.update(symbols)
+        logger.info(f"Subscribed to: {tickers}")
+
+    async def unsubscribe(self, symbols: set[str]):
+        if not self._running:
+            logger.warning("WebSocket is not running.")
+            return
+
+        if self._client is None:
+            raise RuntimeError("WebSocketClient is None during unsubscribe.")
+
+        if not symbols:
+            logger.warning("No symbols to unsubscribe.")
+            return
+
+        tickers = [f"AM.{symbol}" for symbol in symbols]
+        try:
+            self._client.unsubscribe(*tickers)
+            self._symbols.difference_update(symbols)
+            logger.info(f"Unsubscribed from: {tickers}")
+        except ConnectionClosedError as e:
+            logger.warning(f"WebSocket connection was already closed: {e}")
+
+    async def handle_messages(self, msgs: list[WebSocketMessage]):
+        logger.info(f"Received messages: {msgs}")
+        pushed = 0
+        for msg in msgs:
+            if not isinstance(msg, EquityAgg):
+                logger.warning("Received unexpected message %s", str(msg))
+                continue
+
+            record = {
+                "symbol": msg.symbol,
+                "timestamp": msg.start_timestamp,
+                "open": msg.open,
+                "high": msg.high,
+                "low": msg.low,
+                "close": msg.close,
+                "volume": msg.volume,
+            }
+            channel_id = f"candles_min_{msg.symbol}"
+            await self._channel.push(channel_id, record, flush=False)
+            pushed += 1
```

Comment:
Not necessary to have a pushed count. You can implement the channel to avoid flushing if there are no entries to flush.

Reply:
I see that you're already checking if self._pipeline is true. That should be enough.